include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -547,7 +547,8 @@ struct __result_and_scratch_storage
     }
 
   public:
-    __result_and_scratch_storage(_ExecutionPolicy& __exec, ::std::size_t __scratch_n)
+    template <typename _Policy>
+    __result_and_scratch_storage(_Policy&& __exec, ::std::size_t __scratch_n)
```

Comment:
What if we write `__scratch_n = 0` and remove `0` as second param in all instantiations?

Reply:
I think I also prefer to keep this as is, to help code using `__result_and_scratch_storage` make it more clear that they are not using the "scratch space" portion of this structure, but I don't have strong opinions on it.  

I'm interested in @julianmi  opinion here also.

Reply:
I tend to agree that the current version (without the default value) is clearer.

Reply:
ok, let's save as is